pkg/policy/identifier.go

```diff
@@ -43,6 +43,12 @@ type Endpoint interface {
 	GetID16() uint16
 	GetSecurityIdentity() *identity.Identity
 	PolicyRevisionBumpEvent(rev uint64)
+	LockAlive() error
+	Unlock()
+	RLockAlive() error
+	RUnlock()
+	UnconditionalLock()
+	UnconditionalRLock()
```

Comment:
This seems to expose a bunch of the locking complexity of the endpoint to the policy package where I think we should try to avoid increasing the complexity. If there are a bunch of areas where the current code really needs these then I think I see a fair argument to add them, but for reading the identity it seems a bit overkill. Do you have a longer term use of these in mind?